lib/llm/OpenAIClient.ts

```diff
@@ -1,27 +1,51 @@
 import OpenAI from "openai";
 import { zodResponseFormat } from "openai/helpers/zod";
 import { LLMClient, ChatCompletionOptions } from "./LLMClient";
+import { LLMCache } from "./LLMCache";
 
 export class OpenAIClient implements LLMClient {
   private client: OpenAI;
+  private cache: LLMCache;
   public logger: (message: {
     category?: string;
     message: string;
     level?: number;
   }) => void;
+  private enableCaching: boolean;
 
   constructor(
     logger: (message: {
       category?: string;
       message: string;
       level?: number;
     }) => void,
+    enableCaching = false,
   ) {
     this.client = new OpenAI();
     this.logger = logger;
+    this.cache = new LLMCache(logger);
+    this.enableCaching = enableCaching;
   }
 
   async createChatCompletion(options: ChatCompletionOptions) {
+    const cacheOptions = {
+      model: options.model,
+      messages: options.messages,
+      temperature: options.temperature,
+      top_p: options.top_p,
+      frequency_penalty: options.frequency_penalty,
+      presence_penalty: options.presence_penalty,
+      image: options.image,
```

Comment:
hashing the image base64 string might be quite memory intensive - any thoughts on that?